src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -64,7 +69,15 @@ public function setPrefix(string $prefix): void
 
     protected function canBeExtracted(string $file): bool
     {
-        return 'php' === pathinfo($file, \PATHINFO_EXTENSION) && $this->isFile($file);
+        $content = file_get_contents($file);
+
+        return 'php' === pathinfo($file, \PATHINFO_EXTENSION)
+            && $this->isFile($file)
+            && ((bool) preg_match(self::T_FUNCTION_REGEX, $content)
+                || (bool) preg_match(self::TRANS_METHOD_REGEX, $content)
+                || (bool) preg_match(self::TRANSLATABLE_MESSAGE_REGEX, $content)
+                || (bool) preg_match(self::CONSTRAINT_VALIDATION_MESSAGE_REGEX, $content))
```

Comment:
let's inline and remove consts + skip regexps:

```php
            && (false !== stripos($content, 't(')
                || false !== stripos($content, '->trans(')
                || false !== stripos($content, 'message'))
```

Reply:
The issue here is that `t(` will match much more other things than `t('foo.bar')` like `foot('bar')`. Same for `message` (btw, the regex does not fix this issue)